test/inductor/test_torchinductor.py

```diff
@@ -4002,6 +4002,27 @@ def forward(x):
         ]
         self.common(forward, args)
 
+    def test_zero_dim_reductions(self):
+        devices = ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+        for device in devices:
+            for kd in [True, False]:
+                inps0 = (torch.zeros(2, 0, device=device, dtype=torch.float16), 1, kd)
+                failed_ops = [aten.argmin, aten.argmax, aten.max, aten.min]
+                for fo in failed_ops:
+                    with self.assertRaisesRegex(
+                        IndexError, "Expected reduction dim 1 to have non-zero size"
+                    ):
+                        mod = make_fx(fo)(*inps0)
```

Comment:
any reason you are not using `torch._dynamo.optimize` here?

Reply:
errr, no good reason, lol, I'll update it

Reply:
okay, I actually remembered, this was a decent reason: if I use `torch._dynamo.optimize`, it breaks `assertRaisesRegex` since it relies on searching a regex in `exc_value` in `__exit__` of `_AssertRaisesContext`.
`torch._dynamo.optimize` wraps around an original exception into `torch._dynamo.exc.InternalTorchDynamoError` and that doesn't set `exc_value`. I'm not sure I really want to investigate how exactly `torch._dynamo.exc.InternalTorchDynamoError` breaks `exc_value` 😛